cmd/soroban-rpc/internal/util/monitoredroutine.go

```diff
@@ -0,0 +1,78 @@
+package util
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"runtime"
+	"runtime/debug"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stellar/go/support/log"
+)
+
+type MonitoredRoutineConfiguration struct {
+	Log                *log.Entry
+	LogPanicsToStdErr  bool
+	ExitProcessOnPanic bool
+	PanicsCounter      prometheus.Counter
+}
+
+// MonitoredRoutine give us the ability to spin a goroutine, with clear upfront definitions on what should be done in the
+// case of an internal panic.
+func MonitoredRoutine(cfg MonitoredRoutineConfiguration, fn func()) {
```

Comment:
Interestingly, inspired by [`errgroup`](https://pkg.go.dev/golang.org/x/sync/errgroup), I noticed that the zero-value for `MonitoredRoutineConfiguration` is valid, so you could do:
```suggestion
func (cfg MonitoredRoutine) Go(fn func()) {
```

Which would let you:
```go
watchdog := MonitoredRoutine{ Log: logger }
watchdog.Go(func() {  })
```
Or maybe there's an abstraction called `panicgroup`... But I'm not sure if that's useful? 🤔